src/puppetlabs/trapperkeeper/services/status/status_proxy_service.clj

```diff
@@ -0,0 +1,22 @@
+(ns puppetlabs.trapperkeeper.services.status.status-proxy-service
+  (:require [clojure.tools.logging :as log]
+    [puppetlabs.trapperkeeper.core :refer [defservice]])
+  (:import (java.net URL)))
+
+(defservice status-proxy-service
+  [[:WebroutingService add-proxy-route]
+   [:ConfigService get-in-config]]
+  (init [this context]
+    (log/info "Initializing status service proxy")
+    (let [target-url (URL. (get-in-config [:status-proxy :proxy-target-url]))
+          host (.getHost target-url)
+          port (.getPort target-url)
+          path (.getPath target-url)]
+      (add-proxy-route
+        this
+        {:host host
+         :port port
+         :path path}
+        {:ssl-config (get-in-config [:status-proxy :ssl-opts])
+         :scheme     :https}))
```

Comment:
Not that anyone's likely to use it this for anything but proxying to an "https" URL, but maybe it would be good to do some sort of validation on this scheme.  Like by doing a `.getProtocol` call on the `target-url` and translate a return value of `http` or `https` to the corresponding keyword else fail startup with some sort of "unsupported scheme" error.